### ISSUE TITLE
upload to F-Droid

### ISSUE THREAD

**Issue by yshrsmz** [yshrsmz/monotweety#42]
https://f-droid.org/contribute/
https://f-droid.org/wiki/page/Inclusion_Policy
https://f-droid.org/wiki/page/Inclusion_How-To

need to exclude firebase and some other libraries?

**Comment by yshrsmz** [comment]
twitter-kit can be used, but need to check if it can be used without Fabric

**Comment by yshrsmz** [comment]
looks like Twitter-Kit is not able to be initialized without Fabric.
So if We want to upload to F-Droid, We need to migrate to another Twitter library :(

**Comment by yshrsmz** [comment]
migrate to Twitter4J?

**Comment by IzzySoft** [comment]
If you could at least attach the `.apk` to the `/releases` (for the current one and future ones), I could add it to [my F-Droid compatible repo](https://android.izzysoft.de/repo). New versions would then be available there within 24h of your attaching them to a new tag/release. The repo can be used with the F-Droid client easily.

**Comment by yshrsmz** [comment]
@IzzySoft Thanks for your suggestion!
But to do so, at least we need to remove all non-OSS dependencies, right?(I'm not entirely sure about F-Droid though)

**Comment by IzzySoft** [comment]
For the "official repo" yes. For my repo it's preferable, but I'd accept it as it is currently. There are already some apps in I recently discovered that they use Firebase (or Google Analytics or even Flurry). I might go marking them as having unfree dependencies (and/or "Tracking you" e.g. for some Analytics modules like Flurry), though. If you browse my repo (using a web browser) you will notice I list the "ingredients" (read: libraries used) for each app, and leave the decision to the user. If something demands a "special mark", I set that (Ads etc.) so it also shows in the Android client. If I see something being a "big violation", I won't accept it (or throw it out if found later), as recently with a dialer "backported from Android N" which had the Umeng (Chinese Flurry) added, but such cases occur rarely. I see my repo as a compromise *complementing* the official one for apps otherwise not available outside of proprietary stores, so rules are less strict (but not non-existent; [details here](https://android.izzysoft.de/repo/info)).

**TL;DR:** As soon as you add the `.apk` to `releases/` I'd add it to my repo. If you remove non-OSS dependencies, that'd be a big plus and much appreciated, but not a prerequisite.

**Comment by yshrsmz** [comment]
@IzzySoft thanks for the explanation. That seems really nice to me. will let you know when it's ready(won't take so long, I think)

**Comment by IzzySoft** [comment]
Any progress so far?

**Comment by yshrsmz** [comment]
@IzzySoft Sorry for being late.
I've just added an apk file to the latest public release.
https://github.com/yshrsmz/monotweety/releases/tag/v1.5.1

**Comment by IzzySoft** [comment]
Thanks a lot! Find it in my repo [here](https://apt.izzysoft.de/fdroid/index/apk/net.yslibrary.monotweety) now (feel free to link to it :wink:).

**Comment by yshrsmz** [comment]
Thank you!
I've added the link in README
